app/src/main/java/com/quran/labs/androidquran/widgets/ForceCompleteTextView.java

```diff
@@ -34,6 +34,10 @@ private void init() {
     super.setOnItemClickListener((parent, view, position, id) -> onForceComplete(position, id));
   }
 
+  public void forceComplete() {
+    onForceComplete(AdapterView.INVALID_POSITION, AdapterView.INVALID_ROW_ID);
+  }
```

Comment:
what if we convert this class to kotlin and set `AdapterView.INVALID_POSITION` and `AdapterView.INVALID_ROW_ID` as default value for `position` & `rowId` in function `onForceComplete`, so we no need to add this `forceComplete` function.

Reply:
What do you thing @ozbek ?

Reply:
Converting to Kotlin aside, I don't think you can call `onForceComplete()` from `JumpFragment` class (at least, not without changing its scope definition).

Reply:
Yes (what I meant was converting to Kotlin needs to be done for other reasons:). What do you say @hakimrie?

Reply:
I think all force complete should be called from Listener (OnForceCompleteListener) to prevent leaky abstraction.
it better to update `jump_dialog.xml` by adding this two attribute in parent layout:
```
android:focusableInTouchMode="true"
android:clickable="true"
```
to make `onFocusChanged` called when user click outside ForceCompleteTextView.
what do you think @fikr4n @ozbek ?

Reply:
Not sure why, but I've tried to set those two properties in several views (trial and error) in `jump_dialog.xml`, but cannot achieve force completion when clicking OK button.

Reply:
@fikr4n I am sorry for very slow response. You're right, It doesn't force completion if user click outside `jump_dialog.xml` layout, in only work if user click inside it. 

I propose we request parent layout focus programmatically before [dismissing](https://github.com/quran/quran_android/blob/3a6888406aff86ba5ada02c83f3881f028e66346/app/src/main/java/com/quran/labs/androidquran/ui/fragment/JumpFragment.java#L155) AlertDialog to make  sure`ForceAutoCompleteTextView.onFocusChanged` called and JumFragment.suraInput has correct value.